### PR TITLE
Update DIM's loadout mods model to Lightfall

### PR DESCRIPTION
**Pull request by robojumper**
Some placeholders left but this removes combat mods, artifact mods, and armor energy type from the loadouts model (i.e. only changes how DIM thinks mods work and how affinity affects them, doesn't remove elemental affinity yet from armor) and adds artifice mods. This is as accurate to the Bungie previews as possible, but there could always be surprises when Lightfall launches.

### CODE REVIEW

**Comment by bhollis** [comment]
Looking great so far!

File: src/app/loadout-builder/process-worker/auto-stat-mod-utils.ts [hunk 1]
```diff
@@ -8,22 +8,22 @@ import { ProcessMod } from './types';
...
-  [StatHashes.Recovery]: { hash: 2645858828, cost: 4, statHash: StatHashes.Recovery },
-  [StatHashes.Discipline]: { hash: 4048838440, cost: 3, statHash: StatHashes.Discipline },
-  [StatHashes.Intellect]: { hash: 3355995799, cost: 5, statHash: StatHashes.Intellect },
-  [StatHashes.Strength]: { hash: 3253038666, cost: 3, statHash: StatHashes.Strength },
+  [StatHashes.Mobility]: { hash: 4183296050, cost: 3, statHash: StatHashes.Mobility },
+  [StatHashes.Resilience]: { hash: 1180408010, cost: 4, statHash: StatHashes.Resilience },
+  [StatHashes.Recovery]: { hash: 4204488676, cost: 4, statHash: StatHashes.Recovery },
+  [StatHashes.Discipline]: { hash: 1435557120, cost: 3, statHash: StatHashes.Discipline },
+  [StatHashes.Intellect]: { hash: 2724608735, cost: 4, statHash: StatHashes.Intellect },
+  [StatHashes.Strength]: { hash: 4287799666, cost: 3, statHash: StatHashes.Strength },
```
Comment (bhollis) [inline src/app/loadout-builder/process-worker/auto-stat-mod-utils.ts:16]: I guess we should generate these from d2ai now?

File: src/app/loadout/known-values.ts [hunk 2]
```diff
@@ -22,4 +25,6 @@ export const activityModPlugCategoryHashes = [
 export const knownModPlugCategoryHashes = [
   ...armor2PlugCategoryHashes,
   ...activityModPlugCategoryHashes,
+  // FIXME add artifice here
+  99999999,
```
Comment (bhollis) [inline src/app/loadout/known-values.ts:29]:
```suggestion
  PlugCategoryHashes.EnhancementsArtifice
```

File: src/app/loadout-builder/process-worker/auto-stat-mod-utils.ts (hunk 1 above)
Comment (robojumper) [reply src/app/loadout-builder/process-worker/auto-stat-mod-utils.ts:16]: Once I sort out the rest of LO yes, this is just to fix the crash with the auto mods toggle

**Review by bhollis — APPROVED** [review]

**Pull request merged**